src/test/java/com/google/impactdashboard/database_manager/DataReadManagerTest.java

```diff
@@ -0,0 +1,147 @@
+package com.google.impactdashboard.database_manager;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.impactdashboard.configuration.Configuration;
+import com.google.impactdashboard.data.project.ProjectIdentification;
+import com.google.impactdashboard.data.recommendation.*;
+import com.google.impactdashboard.database_manager.data_read.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.io.IOException;
+
+@RunWith(JUnit4.class)
+public class DataReadManagerTest {
+
+  private static DataReadManager dataReadManager;
+
+  private static final ProjectIdentification PROJECT_1_IDENTIFICATION = 
+    ProjectIdentification.create("project-1", "project-id-1", 123456789123L);
+  private static final ProjectIdentification PROJECT_2_IDENTIFICATION = 
+    ProjectIdentification.create("project-2", "project-id-2", 234567890123L);
+  private static final String PROJECT_ID_1 = "project-id-1";
+  private static final String PROJECT_ID_2 = "project-id-2";
+  private static final Recommendation PROJECT_2_RECOMMENDATION_ON_25TH = 
+    Recommendation.create(
+      PROJECT_ID_2, "Remove unused permissions from user test5@example.com", 
+      Recommendation.RecommenderType.IAM_BINDING, 1593115512000L, 
+      IAMRecommenderMetadata.create(350));
+  private static final Recommendation PROJECT_2_RECOMMENDATION_ON_9TH = 
+    Recommendation.create(
+      PROJECT_ID_2, "Remove unused permissions from user test4@example.com", 
+      Recommendation.RecommenderType.IAM_BINDING, 1591704613000L, 
+      IAMRecommenderMetadata.create(500));
+
+  @BeforeClass
+  public static void setTestingConfiguration() throws IOException {
+    Configuration.useTestDatabase = true;
+    dataReadManager = DataReadManagerFactory.create();
+  }
+
+  @AfterClass
+  public static void undoTestingConfiguration() {
+    Configuration.useTestDatabase = false;
```

Comment:
is this needed?

Since DataReadManagerFactory returns the same value each time (I think), doesn't setting this flag after the tests have no effect?

Reply:
I'm pretty sure the tests will be run right before deploying the code, which I thought would result in this flag being set to true in the deployed code. `DataReadManagerFactory` would still return the correct thing since it depends on a different testing flag, but `QueryConfigurationBuilderFactory` would return an implementation that queries the test database instead of the real one.